test/Rollup.spec.ts

```diff
@@ -1,377 +1,163 @@
-            alicePDAProof,
-            accountProofs
-        );
-
-        assert.equal(error, ErrorCode.InvalidTokenAmount, "Wrong ErrorId");
-        await utils.compressAndSubmitBatch(tx, newStateRoot);
-        const batchId = await utils.getBatchId();
-
-        falseBatchComb = {
-            batchId,
-            txs: [tx],
-            batchProofs: {
-                accountProofs: [accountProofs],
-                pdaProof: [alicePDAProof]
-            }
+    it("submit a batch and dispute", async function() {
+        const tx = new TxTransfer(
+            Alice.stateID,
+            Bob.stateID,
+            5,
+            Alice.nonce + 1
+        );
+
+        const signature = Alice.sign(tx);
+
+        const rollup = contracts.rollup as Rollup;
+        const rollupUtils = contracts.rollupUtils as RollupUtils;
+        const stateRoot = stateTree.root;
+        const proof = stateTree.applyTxTransfer(tx);
+        const txs = ethers.utils.arrayify(tx.encode(true));
+        const _tx = await rollup.submitBatch(
+            [txs],
+            [stateRoot],
+            Usage.Transfer,
+            [mcl.g1ToHex(signature)],
+            { value: ethers.utils.parseEther(TESTING_PARAMS.STAKE_AMOUNT) }
+        );
+        await _tx.wait();
+
+        const batchId = Number(await rollup.numOfBatchesSubmitted()) - 1;
+        const root = await registry.root();
+        const rootOnchain = await registry.registry.root();
+        assert.equal(root, rootOnchain, "mismatch pubkey tree root");
+        const batch = await rollup.getBatch(batchId);
+
+        const commitment = {
+            stateRoot,
+            accountRoot: root,
+            txHashCommitment: ethers.utils.solidityKeccak256(["bytes"], [txs]),
+            batchType: Usage.Transfer
         };
-    });
-
-    it("submit new batch 7th(false batch)", async function() {
-        const aliceState = stateStore.items[Alice.Path];
-        const tx = {
-            txType: Usage.Transfer,
-            fromIndex: Alice.AccID,
-            toIndex: Bob.AccID,
-            tokenType: Alice.TokenType,
-            amount: 0, // An invalid amount
-            nonce: aliceState.data!.nonce + 1
-        } as Transaction;
-        tx.signature = await utils.signTx(tx, wallets[0]);
-        const {
-            accountProofs,
-            newStateRoot
-        } = await utils.processTransferTxOffchain(stateStore, tx);
-        stateStore.restoreCheckpoint();
-
-        // process transaction validity with process tx
-        const { error } = await utils.processTransferTx(
-            tx,
-            alicePDAProof,
-            accountProofs
+        const depth = 1; // Math.log2(commitmentLength + 1)
+        const tree = Tree.new(
+            depth,
+            Hasher.new(
+                "bytes",
+                ethers.utils.keccak256(
+                    "0x0000000000000000000000000000000000000000000000000000000000000000"
+                )
+            )
+        );
```

Comment:
Here highlighted some behaviors of `MerkleTreeUtils.getMerkleRootFromLeaves`, which we used in rollup contract to get Merkle root of commitments. Might be gotchas for clients.

- It uses `hash(ZERO_BYTES)` as zero leaf.
- It uses minimum possible height, depending on the length of the leaves.